src/qt/winshutdownmonitor.cpp

```diff
@@ -25,7 +27,7 @@ bool WinShutdownMonitor::nativeEventFilter(const QByteArray &eventType, void *pM
            {
                // Initiate a client shutdown after receiving a WM_QUERYENDSESSION and block
                // Windows session end until we have finished client shutdown.
-               StartShutdown();
+               Assert(kernel::g_context)->interrupt();
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/28051#discussion_r1257286292

> I think we might have introduced a bug here before. This may be possible to trigger when running the GUI before the kernel context is initialized with `app.baseInitialize()`. We could just move it to after `baseInitialize` (see my [patch here](https://github.com/TheCharlatan/bitcoin/commit/2df7a669a07be01ef2cce0d22891617bc19060d7)), but I am not sure if that would be correct either. Maybe `app.baseInitialize()` can be moved up a bit instead?

This bug was resolved by moving the signalinterrupt object from the kernel context to the node context and initializing it earlier. Also by abstracting this code and having it call interfaces::Node::startShutdown()